kong/plugins/oauth2/access.lua

```diff
@@ -163,13 +164,13 @@ local function authorize(conf)
       end
 
       -- Check scopes
-      local ok, scopes = retrieve_scopes(parameters, conf)
-      if not ok then
-        response_params = scopes -- If it's not ok, then this is the error message
+      local scopes, err = retrieve_scope(parameters, conf)
+      if err then
+        response_params = err -- If it's not ok, then this is the error message
```

Comment:
now that you changed the return type to `scope | nil, err` this comment (and the similar one below) aren't necessary :)

Reply:
Yes, changed all the `return false` to `return nil`